tests/test_toc.py

```diff
@@ -24,7 +25,9 @@ def test_simple_toc():
 
 
 def test_full_toc():
-    if fitz.mupdf_version_tuple >= (1, 23, 0):
+    if fitz.pymupdf_version_tuple >= (1, 24, 0) and hasattr(fitz, "mupdf"):
```

Comment:
I'm not sure tests should ever look at `pymupdf_version_tuple`. The only reason for doing so would be to allow a git checkout of PyMuPDF to be used to test an older version of PyMuPDF that has been installed locally. This is hardly ever useful i think, and it introduces a potentially complicated extra set of decisions in the test code.

There are a couple of other places where i've incorrectly written test code that looks at `pymupdf_version_tuple`; i will remove these references.

[Having test code look at fitz.`mupdf_version_tuple` is different i think, because it allows us to test the latest PyMuPDF code with MuPDF release and master branches, both of which are useful things to do.]